api/v1alpha1/velero_types.go

```diff
@@ -54,47 +54,70 @@ const CSIPluginImageKey UnsupportedImageKey = "csiPluginImageFqin"
 	// +optional
-	CustomVeleroPlugins []CustomPlugin `json:"customVeleroPlugins,omitempty"`
+	CustomPlugins []CustomPlugin `json:"customPlugins,omitempty"`
+	// RestoreResourceVersionPriority represents a configmap that will be created if defined for use in conjunction with `EnableAPIGroupVersions` feature flag
+	// Defining this field automatically add EnableAPIGroupVersions to the velero server feature flag
+	// +optional
+	RestoreResourcesVersionPriority string `json:"restoreResourcesVersionPriority,omitempty"`
+	// If you need to install Velero without a default backup storage location NoDefaultBackupLocation flag is required for confirmation
+	// +optional
+	NoDefaultBackupLocation bool `json:"noDefaultBackupLocation,omitempty"`
+}
 
-	// EnableRestic is a boolean to specify if restic daemonset instance should be created or not
+type ResticConfig struct {
+	// SupplementalGroups defines the linux groups to be applied to the Restic Pod
 	// +optional
-	EnableRestic *bool `json:"enableRestic,omitempty"`
-	// ResticSupplementalGroups defines the linux groups to be applied to the Restic Pod
+	SupplementalGroups []int64 `json:"supplementalGroups,omitempty"`
+	// NodeSelector defines the nodeSelector to be supplied to Restic podSpec
 	// +optional
-	ResticSupplementalGroups []int64 `json:"resticSupplementalGroups,omitempty"`
-	// ResticNodeSelector defines the nodeSelector to be supplied to Restic podSpec
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// Tolerations defines the list of tolerations to be applied to Restic daemonset
 	// +optional
-	ResticNodeSelector map[string]string `json:"resticNodeSelector,omitempty"`
-	// ResticTolerations defines the list of tolerations to be applied to Restic daemonset
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// ResourceAllocations defines the CPU and Memory resource allocations for the restic Pod
 	// +optional
-	ResticTolerations []corev1.Toleration `json:"resticTolerations,omitempty"`
-	// ResticResourceAllocations defines the CPU and Memory resource allocations for the restic Pod
+	ResourceAllocations corev1.ResourceRequirements `json:"resourceAllocations,omitempty"`
+	// Timeout defines the Restic timeout, default value is 1h
 	// +optional
-	ResticResourceAllocations corev1.ResourceRequirements `json:"resticResourceAllocations,omitempty"`
-	// ResticTimeout defines the Restic timeout, default value is 1h
+	Timeout string `json:"timeout,omitempty"`
+}
+
+type ServerConfig struct {
+	*VeleroConfig
+	*ResticConfig
+}
+
+type BackupStorage struct {
+	*velero.BackupStorageLocationSpec
+}
+
+type VolumeSnapshot struct {
+	*velero.VolumeSnapshotLocationSpec
+}
+
+// VeleroSpec defines the desired state of Velero
+type BackupApplicationSpec struct {
+	// BackupStorageLocations defines the list of desired configuration to use for BackupStorageLocations
+	// +optional
+	BackupStorageLocations []BackupStorage `json:"backupStorageLocations"`
```

Comment:
NIT: since we are creating a new type `BackupStorage`, should we consider moving away from naming the field `backupStorageLocations`? `spec.backupStorageLocations` feels very close to a velero storage location API I wonder when we have our own configurations in the future if we would want it to be named like this.